new_wordcount_and_median.py

```diff
@@ -0,0 +1,64 @@
+import os
```

Comment:
This version isn't especially better, since all of the functions have an almost identical implementation.  The main advantage is that instead of having to keep passing our lists and dictionaries around as function parameters, we can just jam them into ``self`` (the instance of the class as seen by the class methods) and refer to them whenever it's convenient.